devito/arch/compiler.py

```diff
@@ -472,6 +474,37 @@ def __lookup_cmds__(self):
         self.MPICXX = 'mpicxx'
 
 
+class CrayCompiler(Compiler):
```

Comment:
That's a lot of duplicates cannot it just inherit from the amd compiler? Also my understanding is that aompcc is getting deprecated and should switch to amdclang with all the flags always

Reply:
it would be conceptually bad to inherit from AOMPCompiler

I think I'm OK with a certain degree of duplication here

What's the CRAY compiler based on underneath? gnu or clang?

Reply:
but yeah, regardless, in my experience these compiler classes tend to require too much special-casing, so it's in general difficult and not particularly simplifying to share code in some more abstract base classes... I think

Reply:
It is clang underneath. I will see what I can drop, this is why I marked the PR as draft.
I will explore these options, and will ping you on any more advice

Reply:
>it would be conceptually bad to inherit from AOMPCompiler

Yes but the way it is going with aomp, both this one and aomp should be a `ClangCompiler` with their few specific flags

Reply:
So I think we can inherit from Clang so far; I do not see any difference apart from the compiler invocation name.